


![](img/add_cell.png)



To make a new cell you can click on the **+** in the toolbar above.

To change the cell to a text cell, you can click on the ***Code*** dropdown menu above and change it to ***Markdown***.

Delete this cell (by right clicking and selecting ***Delete Cells***) before handing in your project. 

Data Analysis for the Disney Movies

Introduction : 

In this analysis there will be analysis of various question associated with the collection of Disney movies.I am interested in finding out regarding the Disney movies dataset.

Methods and Results

I am  interested in computing the Disney Movies collection on the basis of the revenue as I will be conducting my analysis. This implies that I will only be using only **disney movies total gross** tables as that is only tables which has a relevant data for analysis.

However, before moving further, let us import the libraries and tables.

In [1]:
import altair as alt
import pandas as pd

In [2]:
disney_movies = pd.read_csv('data/disney_movies_total_gross.csv')
disney_movies

movie_title  release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                          Pinocchio   Feb 9, 1940  Adventure           G   
2                           Fantasia  Nov 13, 1940    Musical           G   
3                  Song of the South  Nov 12, 1946  Adventure           G   
4                         Cinderella  Feb 15, 1950      Drama           G   
..                               ...           ...        ...         ...   
574         The Light Between Oceans   Sep 2, 2016      Drama       PG-13   
575                   Queen of Katwe  Sep 23, 2016      Drama          PG   
576                   Doctor Strange   Nov 4, 2016  Adventure       PG-13   
577                            Moana  Nov 23, 2016  Adventure          PG   
578     Rogue One: A Star Wars Story  Dec 16, 2016  Adventure       PG-13   

      total_gross inflation_adjusted_gross  
0    $184,925,485           $5,228,953,251  
1     $84,300,000           $2,188,229,052  
2     $83,320,000           $2,187,090,808  
3     $65,000,000           $1,078,510,579  
4     $85,000,000             $920,608,730  
..            ...                      ...  
574   $12,545,979              $12,545,979  
575    $8,874,389               $8,874,389  
576  $232,532,923             $232,532,923  
577  $246,082,029             $246,082,029  
578  $529,483,936             $529,483,936  

[579 rows x 6 columns]

In [3]:
print(disney_movies.head(579))
for columns in disney_movies.total_gross:
    disney_movies['total_gross'] = disney_movies['total_gross'].str.replace(r'\W',"",regex=True)

                         movie_title  release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                          Pinocchio   Feb 9, 1940  Adventure           G   
2                           Fantasia  Nov 13, 1940    Musical           G   
3                  Song of the South  Nov 12, 1946  Adventure           G   
4                         Cinderella  Feb 15, 1950      Drama           G   
..                               ...           ...        ...         ...   
574         The Light Between Oceans   Sep 2, 2016      Drama       PG-13   
575                   Queen of Katwe  Sep 23, 2016      Drama          PG   
576                   Doctor Strange   Nov 4, 2016  Adventure       PG-13   
577                            Moana  Nov 23, 2016  Adventure          PG   
578     Rogue One: A Star Wars Story  Dec 16, 2016  Adventure       PG-13   

      total_gross inflation_adjusted_gross  
0    $184,925,485           $5

In [4]:
print(disney_movies.head(579))
for columns in disney_movies.inflation_adjusted_gross:
    disney_movies['inflation_adjusted_gross'] = disney_movies['inflation_adjusted_gross'].str.replace(r'\W',"",regex=True)

                         movie_title  release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs  Dec 21, 1937    Musical           G   
1                          Pinocchio   Feb 9, 1940  Adventure           G   
2                           Fantasia  Nov 13, 1940    Musical           G   
3                  Song of the South  Nov 12, 1946  Adventure           G   
4                         Cinderella  Feb 15, 1950      Drama           G   
..                               ...           ...        ...         ...   
574         The Light Between Oceans   Sep 2, 2016      Drama       PG-13   
575                   Queen of Katwe  Sep 23, 2016      Drama          PG   
576                   Doctor Strange   Nov 4, 2016  Adventure       PG-13   
577                            Moana  Nov 23, 2016  Adventure          PG   
578     Rogue One: A Star Wars Story  Dec 16, 2016  Adventure       PG-13   

    total_gross inflation_adjusted_gross  
0     184925485           $5,228

As the dataset above is not cleaned and had dollar($) and comma(,) symbol inserted in the total_gross and inflation_adjusted_gross and had to remove those values which will help to sort data in ascending and the descending order to evaluate which movie has highest totalgross and lowest totalgross respectively based on the condition.

In [9]:
disney_movies.dtypes

movie_title                         object
release_date                datetime64[ns]
genre                               object
MPAA_rating                         object
total_gross                          int64
inflation_adjusted_gross             int64
dtype: object

In [10]:
disney_movies['release_date'] = pd.to_datetime(disney_movies.release_date)
disney_movies

movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

     total_gross  inflation_adjusted_gross  
0      184925485                5228953251  
1       84300000                2188229052  
2       83320000                2187090808  
3       65000000                1078510579  
4       85000000                 920608730  
..           ...                       ...  
574     12545979                  12545979  
575      8874389                   8874389  
576    232532923                 232532923  
577    246082029                 246082029  
578    529483936                 529483936  

[579 rows x 6 columns]

In [11]:
disney_movies['total_gross'] = disney_movies['total_gross'].astype(int)
disney_movies['inflation_adjusted_gross'] = disney_movies['inflation_adjusted_gross'].astype(int)

In [12]:
disney_movies.dtypes

movie_title                         object
release_date                datetime64[ns]
genre                               object
MPAA_rating                         object
total_gross                          int64
inflation_adjusted_gross             int64
dtype: object

The release_date,total_gross and inflation_adjusted_gross is not in a proper dtypes so conveted in a proper dtypes which will help to sort data based on the given condition

In [13]:
disney_df = disney_movies.sort_values('total_gross',ascending = False)
disney_df

movie_title release_date      genre  \
564    Star Wars Ep. VII: The Force Awakens   2015-12-18  Adventure   
524                            The Avengers   2012-05-04     Action   
578            Rogue One: A Star Wars Story   2016-12-16  Adventure   
571                            Finding Dory   2016-06-17  Adventure   
558                 Avengers: Age of Ultron   2015-05-01     Action   
..                                      ...          ...        ...   
511                                Zokkomon   2011-04-22  Adventure   
20   The Many Adventures of Winnie the Pooh   1977-03-11        NaN   
29                                Condorman   1981-08-07     Action   
355                     Frank McKlusky C.I.   2002-01-01        NaN   
27                                      Amy   1981-03-20      Drama   

    MPAA_rating  total_gross  inflation_adjusted_gross  
564       PG-13    936662225                 936662225  
524       PG-13    623279547                 660081224  
578       PG-13    529483936                 529483936  
571          PG    486295561                 486295561  
558       PG-13    459005868                 459005868  
..          ...          ...                       ...  
511          PG         2815                      2984  
20          NaN            0                         0  
29          NaN            0                         0  
355         NaN            0                         0  
27          NaN            0                         0  

[579 rows x 6 columns]

Now that we have cleaned and have sorted the data based on the highest total_gross revenue earned by each Disney film and as it was obvious that Star Wars was the highest earning movie  

In [14]:
disney = disney_df['total_gross'].astype('int').sum()
disney

37462335553

As the data was already cleaned above we need to find out what was the total gross revenue was earned by disney movies from  1937 until 2016 during this period.

In [15]:
disney0= disney_df['total_gross'].astype('int').mean()
disney0

64701788.51986183

We need to find out the mean for disney movies to know which movies has performed better and vice-versa  

In [27]:
disney1 = disney_df[(disney_df['total_gross']>=disney0)]
disney1

movie_title release_date                genre  \
564    Star Wars Ep. VII: The Force Awakens   2015-12-18            Adventure   
524                            The Avengers   2012-05-04               Action   
578            Rogue One: A Star Wars Story   2016-12-16            Adventure   
571                            Finding Dory   2016-06-17            Adventure   
558                 Avengers: Age of Ultron   2015-05-01               Action   
..                                      ...          ...                  ...   
57                                 Stakeout   1987-08-05               Comedy   
318              Deuce Bigalow: Male Gigolo   1999-12-10               Comedy   
442                                 Step Up   2006-08-11                Drama   
464  Hannah Montana/Miley Cyrus: Best of B…   2008-02-01  Concert/Performance   
3                         Song of the South   1946-11-12            Adventure   

    MPAA_rating  total_gross  inflation_adjusted_gross  
564       PG-13    936662225                 936662225  
524       PG-13    623279547                 660081224  
578       PG-13    529483936                 529483936  
571          PG    486295561                 486295561  
558       PG-13    459005868                 459005868  
..          ...          ...                       ...  
57            R     65673233                 141592160  
318           R     65535067                 106921374  
442       PG-13     65328121                  84078780  
464           G     65281781                  76646993  
3             G     65000000                1078510579  

[168 rows x 6 columns]

As we found out that 168  movies had performed better than the average and remainning all the movies were below average. 

In [28]:
disney2 = disney_df['genre']
disney3 = disney2.value_counts()
disney3

Comedy                 182
Adventure              129
Drama                  114
Action                  40
Thriller/Suspense       24
Romantic Comedy         23
Musical                 16
Documentary             16
Western                  7
Horror                   6
Black Comedy             3
Concert/Performance      2
Name: genre, dtype: int64

In [29]:
disney4 = disney1['genre']
disney5 = disney4.value_counts() 
disney5

Adventure              65
Comedy                 44
Drama                  20
Action                 17
Thriller/Suspense       7
Romantic Comedy         6
Musical                 6
Western                 2
Concert/Performance     1
Name: genre, dtype: int64

As the above 2 tables reflect data about various genre types and how many movies are there in each type in a dataset. The 1st table reflects all the movies category which has performed below and also includes those movies which has performed better than the average and in the second table it includes only those genre movies which has performed better than the average.

In [19]:
disney6 = pd.DataFrame(disney_df.groupby('genre')['total_gross'].mean().sort_values(ascending=True)).astype('int64')
disney6 = disney6.reset_index()
disney6

genre  total_gross
0           Documentary     11292851
1                Horror     14511478
2          Black Comedy     32514404
3                 Drama     36026078
4                Comedy     44613294
5       Romantic Comedy     50095950
6               Western     51287351
7   Concert/Performance     51728233
8     Thriller/Suspense     58616938
9               Musical     72330259
10               Action    104614082
11            Adventure    127047050

In [20]:
disney7 = pd.DataFrame(disney1.groupby('genre')['total_gross'].mean().sort_values(ascending=True)).astype('int64')
disney7 = disney7.reset_index()
disney7

genre  total_gross
0  Concert/Performance     65281781
1              Western     78294423
2                Drama    102004295
3               Comedy    111116184
4      Romantic Comedy    120102820
5    Thriller/Suspense    140103724
6              Musical    141266983
7               Action    209918484
8            Adventure    223600118

As the above 2 tables reflect data about various genre types and which genre type movies in a dataset has earned how much total gross revenue and what is the mean of those movies. 
The 1st table reflects all the movies category which has performed below and also includes those movies which has performed better than the average and in the second table it includes only those genre movies which has performed better than the average and 3 genre movies type were not better than the average as well.

In [21]:
disney8 = (
    alt.Chart(disney6,width = 500, height = 300)
    .mark_bar().encode(
        x = alt.X("genre:O",title = "Movies Type",sort="-y"),
        y=alt.Y("total_gross:Q", title="Gross Revenue"),
    )
    .properties(title = "Revenue by Movies Type")
)
disney8

alt.Chart(...)

In [22]:
disney9 = (
    alt.Chart(disney7,width = 500, height = 300)
    .mark_bar().encode(
        x = alt.X("genre:O",title = "Movies Type",sort="-y"),
        y=alt.Y("total_gross:Q", title="Gross Revenue"),
    )
    .properties(title = "Revenue by Movies Type")
)
disney9

alt.Chart(...)

Ref
https://stackoverflow.com/questions/66603854/futurewarning-the-default-value-of-regex-will-change-from-true-to-false-in-a-fu